### PR TITLE
fix: reduce api requests on chart view dashboard tile

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #11371 

### Description:

When viewing a non-Table viz dashboard tile, we don't need to get its results because the `ChartView` will deal with it. 
When viewing a table, we can get the results. 

This PR:
- Removed `chart-and-results` endpoint since it's unnecessary now and we reuse other utils from existing hooks. 
- Fetch results job only for `tables`
- Also fixed a width issue with a SQL Table dashboard tile, it's looking good now - was being cropped by overflow.
- Removed `data` prop from `ChartView` wasn't being used anywhere

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/sqlRunner/hooks/useDashboardSqlChart.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    savedSqlUuid: string;
+}): Promise<{ resultsAndColumns: ResultsAndColumns; chart: SqlChart }> => {
+    const chart = await fetchSavedSqlChart({
+        projectUuid,
+        uuid: savedSqlUuid,
+    });
+
+    const hasResultsHandledInChart = chart.config.type !== ChartKind.TABLE;
+
+    if (hasResultsHandledInChart) {
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/hooks/useDashboardSqlChart.tsx:21]: this is where we skip the results job @owlas

**Comment by rephus** [comment]
Everything works as expected , I'll wait until the scroll on tables has been fixed and test it again

**Comment by rephus** [comment]
This is probably out of scope, but if I add the same chart multiple times (same query) the request will be duplicated on the backend with the same query body. We should probably cache on react-query or similar somehow. 

This is probably a minor issue, since not many people will add the same chart , even if they do, they will probably contain different filters in the future, so it makes so make a different request. 

image

**Review by rephus — APPROVED** [review]
I can see scrollbars now. 

horizontal scrollbar only shows at the bottom. We can fix it on a separate PR 

[Screencast from 04-09-24 12:04:26.webm](https://github.com/user-attachments/assets/1aa72aa4-6903-443f-8f94-adbc68cb66a9)

**Pull request merged**